spec/models/user_spec.rb

```diff
@@ -5,10 +5,54 @@
   it { should have_many(:answers) }
   it { should have_many(:awards) }
   it { should have_many(:authorizations).dependent(:destroy) }
+  it { should have_many(:subscriptions).dependent(:destroy) }
 
   it { should validate_presence_of :email }
   it { should validate_presence_of :password }
 
+  context 'subscriptions' do
+    let!(:question) { create(:question) }
+    let(:user) { create(:user) }
+    let(:subscription) { user.subscribe!(question) }
+
+    describe '#subscribe!' do
+      it 'creates subscription' do
+        expect { user.subscribe!(question) }.to change(Subscription, :count).by(1)
+      end
+
+      it 'attributes must match' do
+        expect(subscription.user.id).to eq user.id
+        expect(subscription.question.id).to eq question.id
+      end
+    end
+
+    describe '#unsubscribe!' do
+      let!(:subscription) { create(:subscription, user: user, question: question) }
+
+      it 'deletes subscription' do
+        expect { user.unsubscribe!(question) }.to change(Subscription, :count).by(-1)
+      end
+
+      it 'attributes must match' do
+        expect(subscription.user.id).to eq user.id
+        expect(subscription.question.id).to eq question.id
+      end
+    end
+
+    describe '#subscribed?' do
+      let!(:subscription) { create(:subscription, user: user, question: question)}
+      let(:another_question) { create(:question) }
+
+      it 'user have subscription' do
+        expect(user.subscribed?(question)).to be_truthy
+      end
+
+      it 'user to not have another subscription' do
+        expect(user.subscribed?(another_question)).to be_falsy
```

Comment:
https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/predicate-matchers